pkg/cmd/roachtest/roachtestutil/mixedversion/runner.go

```diff
@@ -242,6 +243,13 @@ func (tr *testRunner) runSingleStep(ctx context.Context, ss *singleStep, l *logg
 	defer func() {
 		prefix := fmt.Sprintf("FINISHED [%s]", timeutil.Since(start))
 		tr.logStep(prefix, ss, l)
+		annotation := fmt.Sprintf("(%d): %s", ss.ID, ss.impl.Description())
+		_, err := tr.cluster.AddGrafanaAnnotation(tr.ctx, tr.logger, false /*internal*/, grafana.AddAnnotationRequest{
```

Comment:
Nice! Do you have a screenshot of how this looks for the whole (test) run?

Reply:
<img width="1920" alt="image" src="https://github.com/cockroachdb/cockroach/assets/56486370/ff9b3ba5-a608-4108-bbe4-636ba02cbaf1">


Not sure how soon the data gets GC, but [the grafana link](https://grafana.testeng.crdb.io/d/crdb-console-sql-byname/crdb-console-sql-by-test-run-and-name?from=1710181540655&to=1710185140655&var-DS_PROMETHEUS=Prometheus&var-test_run_id=darrylwong-1710182316&var-test_name=follower-reads-mixed-version-single-region&var-cluster=darrylwong-1710182316-01-n4cpu2&orgId=1) in case you see this before it does.

Reply:
> Not sure how soon the data gets GC, but [the grafana link](https://grafana.testeng.crdb.io/d/crdb-console-sql-byname/crdb-console-sql-by-test-run-and-name?from=1710181540655&to=1710185140655&var-DS_PROMETHEUS=Prometheus&var-test_run_id=darrylwong-1710182316&var-test_name=follower-reads-mixed-version-single-region&var-cluster=darrylwong-1710182316-01-n4cpu2&orgId=1) in case you see this before it does.

Thanks. By default annotations are kept forever [1], and we're using the default. I'll add the following settings to make sure we don't forget to GC,

```
[annotations.dashboard]
max_age = 3M

[annotations.api]
max_age = 3M
```

[1] https://github.com/grafana/grafana/blob/608b40b2a8860c63f37fe5ac542b2a52b7109317/conf/sample.ini#L1241